sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -603,29 +610,30 @@ namespace Azure { namespace Core { namespace Http {
     // ===== Methods used to build HTTP response =====
 
     /**
-     * @brief Add header to the HTTP response.
+     * @brief Add header to the HTTP response. Returns 0 if header is successfully added, otherwise
+     * this method will return the index of the invalid character from \p name
      *
      * @param name HTTP response header name.
      * @param value HTTP response header value.
      */
-    void AddHeader(std::string const& name, std::string const& value);
+    size_t AddHeader(std::string const& name, std::string const& value);
 
     /**
      * @brief Add header to the HTTP response.
      *
-     * @param header HTTP response header in RFCnnnn format (`OWS header-value OWS`).
+     * @param header HTTP response header in RFC7230 format (`OWS header-value OWS`).
```

Comment:
Use latest RFC? https://tools.ietf.org/html/rfc8615
```suggestion
     * @param header HTTP response header in RFC 8615 format (`OWS header-value OWS`).
```

Reply:
why? 
Is there something we want from 8615?

Reply:
We should be supporting the superseded/most up-to-date RFC, unless there is a reason not to.

Reply:
there is nothing new for headers in 8615 that is not in 7230

Reply:
Then is there any harm in listing the latest? Usually when RFCs is updated, it is good practice to use the latest one available at the point of shipping a product.

Reply:
```suggestion
     * @param header HTTP response header in RFC 7230 format (`OWS header-value OWS`).
```

Reply:
yes, we should reference the one that provides the documentation about what we are using.
7230 defines the latest standard about most of the thing currently used.